evalml/automl/automl_search.py

```diff
@@ -275,6 +270,91 @@ def __init__(self,
                                                    n_splits=3, shuffle=True, random_seed=self.random_seed)
         self.data_splitter = self.data_splitter or default_data_splitter
         self.pipeline_parameters = pipeline_parameters if pipeline_parameters is not None else {}
+        self.search_iteration_plot = None
+        self._interrupted = False
+
+        self._engine = SequentialEngine(self,
+                                        should_continue_callback=self._should_continue,
+                                        pre_evaluation_callback=self._pre_evaluation_callback,
+                                        post_evaluation_callback=self._post_evaluation_callback)
+        self._engine.set_data(self.X_train, self.y_train)
```

Comment:
Setting the data as a step after the constructor is unnecessary right now, but I think it'll be a useful API for the parallel impl, so leaving in.